### PR TITLE
Update to cardano-api 8.36

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
This is the version used by cardano-node 8.7.2

### CODE REVIEW

**Comment by sjoerdvisscher** [comment]
@MaximilianAlgehed If it's ok with you I'd like to put this version on CHaP.

**Comment by MaximilianAlgehed** [comment]
Sounds good to me

**Comment by sjoerdvisscher** [comment]
@MaximilianAlgehed Could you approve the PR too then? 😄

**Review by MaximilianAlgehed — APPROVED** [review]

**Comment by sjoerdvisscher** [comment]
@MaximilianAlgehed Sorry, just pushed another commit to also support GHC 9.6. Can you check that too?

File: quickcheck-contractmodel/src/Test/QuickCheck/ContractModel/Internal.hs [hunk 1]
```diff
@@ -108,8 +109,8 @@ instance (Monad m, HasChainIndex m) => HasChainIndex (RunMonad m) where
   getChainState = lift getChainState
 
 instance (Monad m, HasChainIndex m) => HasChainIndex (StateModel.PostconditionM m) where
-  getChainIndex = lift getChainIndex
-  getChainState = lift getChainState
+  getChainIndex = StateModel.PostconditionM (lift getChainIndex)
```
Comment (MaximilianAlgehed) [inline quickcheck-contractmodel/src/Test/QuickCheck/ContractModel/Internal.hs:112]: @sjoerdvisscher why is this necessary?

Comment (sjoerdvisscher) [reply quickcheck-contractmodel/src/Test/QuickCheck/ContractModel/Internal.hs:112]: The current version of `quickcheck-dynamic` on Hackage doesn't have this instance. It's now fixed on `main` but it is not on Hackage yet.

Comment (MaximilianAlgehed) [reply quickcheck-contractmodel/src/Test/QuickCheck/ContractModel/Internal.hs:112]: 🤦 of course... I remember when that happened... We should probably fix that over in `quickcheck-dynamic` soon...
